components/elasticsearch/scripts/start.py

```diff
@@ -8,11 +11,50 @@
     join('components', 'utils.py'),
     join(dirname(__file__), 'utils.py'))
 import utils  # NOQA
-ES_SERVICE_NAME = 'elasticsearch'
 
+ES_SERVICE_NAME = 'elasticsearch'
 ctx_properties = utils.ctx_factory.get(ES_SERVICE_NAME)
-ES_ENDPOINT_IP = ctx_properties.get('es_endpoint_ip')
+ES_ENDPOINT_IP = ctx_properties['es_endpoint_ip']
+ES_ENDPOINT_PORT = ctx_properties['es_endpoint_port']
+
+
+def _examine_status_response(response):
+    """Check if the status response from elasticsearch is correct.
+
+    A correct response has a status of 200, and also returns a JSON object
+    with {'status': 200}.
+    Elasticsearch doesn't start up immediately, so this retries a few times.
+    """
+    if response.code != 200:
+        return False
+
+    parsed_response = json.load(response)
+
+    return parsed_response['status'] == 200
+
+
+def check_index_exists(url, index_name='cloudify_storage'):
```

Comment:
I'm wondering - do you think there's a reason to use the retry decorator here? Obviously, you first check that an http response is available, so there's no reason why this wouldn't work the first time. Thing is, I'm not sure we can account for any potential problem here. What do you think?